ppa/archive/tests/test_admin.py

```diff
@@ -46,3 +48,15 @@ class Meta:
         digadmin.save_related(request, form, [], False)
         # mocked index method of the digwork object should have been called
         digwork.index.assert_called_with(params={'commitWithin': 10000})
+
+    def test_bulk_add_collection(self):
+        # create a DigitizedWorkAdmin object
+        digworkadmin = DigitizedWorkAdmin(DigitizedWork, AdminSite())
+        fakerequest = Mock()
+        # mock items 1,2,3 being selected
+        fakerequest.POST.getlist.return_value = ['1', '2', '3']
+        redirect = digworkadmin.bulk_add_collection(fakerequest, [])
+        # should return a redirect
+        assert isinstance(redirect, HttpResponseRedirect)
+        # url should reverse the appropriate route and append ?ids=1,2,3
+        assert redirect.url == '%s?ids=1,2,3' % reverse('archive:bulk-add')
```

Comment:
Any error cases that should be tested here?

Reply:
Given the new framework, I think there likely are not. I handled the error of no pks or bad pks (somehow from a previously used url), but the latter is broadly fixed using the session framework (much harder for a user to set them) and the former I handle in the add view (which displays a message if pks haven't been set).

I will add a test for an empty queryset, though. I'm not sure it would pose an issue, but I see no harm in making sure it behaves as expected.